### ISSUE TITLE
Panic on invalid SSL certificate

### ISSUE THREAD

**Issue by yolio2003** [denoland/deno#19520]
```
app.listen({
  port: 2003,
  secure: true,
  certFile: './local/ssl.crt',
  keyFile: './local/ssl.key',
})
```

```
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos aarch64
Version: 1.34.2
Args: ["/opt/homebrew/bin/deno", "run", "--allow-net", "--allow-read", "--allow-write", "--watch", "./local/index.ts"]

thread 'main' panicked at 'invalid key or certificate: General("invalid private key")', /private/tmp/deno-20230609-9231-3igdr4/deno/ext/net/ops_tls.rs:1060:6
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: deno_net::ops_tls::op_net_listen_tls::v8_func
   4: deno_net::ops_tls::op_net_listen_tls::v8_fn_ptr
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by bartlomieju** [comment]
We shouldn't panic in this case, but return a graceful error.

**Comment by RishabhJain-28** [comment]
Hey! I would like to work on this issue.

**Comment by jakecorrenti** [comment]
I'll give this a shot

**Comment by fbaltor** [comment]
Hey, @yolio2003, but what is this `app` in the code example? I couldn't find any `listen` methods with the mentioned signature in the `Deno` namespace or in the `deno_std`...

**Comment by 0xIchigo** [comment]
Fixed with PR #20157 :fire:
